pkgs/development/web/nodejs/node-npm-build-npm-package-logic-node16.patch

```diff
@@ -0,0 +1,95 @@
+This patch is based off of npm tag v8.19.4.
+
+This introduces fixes for 4 issues:
+
+1. When node-gyp is included as a dependency in a project, any scripts that run it will not use the copy included in Node. This is problematic because we patch node-gyp to work without xcbuild on Darwin, leading to these packages failing to build with a sandbox on Darwin.
```

Comment:
> leading to these packages failing to build with a sandbox on Darwin.

We have sandbox on Darwin, now?!

Reply:
Well, something like that. We've had it a while, but it's jank and a lot of packages get an error like `sandbox-exec: pattern serialization length 70491 exceeds maximum (65535)` with it

Also Apple has deprecated the mechanism it uses with no real replacement (though it does still otherwise work just fine)